enable ONE auth file authentication

YouhuaLi opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Now the module supports only username/password authentication. I want to enable authentication by one auth file (~/.one/one_auth)
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Feature Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
opennebula
##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->
python-OCA [support auth file](https://github.com/python-oca/python-oca/blob/0dfd181ee44d4ea05466ca4ed814d290c2c6fcfa/oca/__init__.py#L69) already, so try to make  minimum changes here.

YouhuaLi committed:
enable ONE auth file authentication

YouhuaLi committed:
enable ONE auth file authentication

vmpr committed:
changed documented default value for url (#49097)

the Default Value for url is 3000, so there was a mistake in the documentation

samccann committed:
Gather subset fix (#49264)

*  reword subset description for ios_facts

yungezz committed:
new module: azure_rm_rediscache (#47914)

* add new redis cache module

* fix version

* fix lint

mikewiebe committed:
Fix n3k power supply facts (#49150)

OlivierB committed:
Fix nxos modules idempotence issue (#46437)

* nxos_interface vlan and port-channel idempotence fix for mtu

* Fix MTU reconfiguration at each execution

* nxos_interface port-channel idempotence fix for mode (#44248)

* Fix trunk mode idempotence for port-channel
* Gather ethernet and port-channel code for mode management

* nxos_linkagg port-channel idempotence fix for channel-group's mode

The regex to retrieve channel-group's mode is not enough accurate.
Therefore, the swhitchport mode  was matched instead of the
channel-group's mode.

* This fix add accuracy to match the right configuration command

* Add support for switchport mode dot1q-tunnel in nxos_interface

* Fix layer reconfiguration at each execution

ilicmilan commented on lib/ansible/modules/cloud/opennebula/one_vm.py:
```diff
@@ -52,6 +52,9 @@
     description:
       - Password of the user to login into OpenNebula RPC server. If not set
       - then the value of the C(ONE_PASSWORD) environment variable is used.
+      - if either api_username or api_password is not set, then it will try
```
typo

ganeshrn committed:
Fix iosxr netconf plugin response namespace (#49238)

* Fix iosxr netconf plugin response namespace

*  iosxr netconf plugin removes namespace by default
   for all the responses as parsing of xml is easier
   without namepsace in iosxr module. However to validate
   the response received from device against yang model requires
   namespace to be present in resposne.
*  Add a parameter in iosxr netconf plugin to control if namespace
   should be removed from response or not.

* Fix CI issues

* Fix review comment

ilicmilan commented on lib/ansible/modules/cloud/opennebula/one_vm.py:
```diff
@@ -52,6 +52,9 @@
       - then the value of the C(ONE_PASSWORD) environment variable is used.
+      - if either api_username or api_password is not set, then it will try
+      - authenticate with ONE auth file. Default path is "~/.one/one_auth".
+      - Set environment variable C(ONE_AUTH) to override this path.
```
You forgot to implement this feature

dalaoli-shuxue replied:
it's not necessary but python-oca will do [that](https://github.com/python-oca/python-oca/blob/0dfd181ee44d4ea05466ca4ed814d290c2c6fcfa/oca/__init__.py#L70) if you pass None to the 'secret' argument.  No need to duplicate the work here.

YouhuaLi replied:
It's not necessary because python-oca will do [that](https://github.com/python-oca/python-oca/blob/0dfd181ee44d4ea05466ca4ed814d290c2c6fcfa/oca/__init__.py#L69) if you pass None as the 'secret' argument. And it's impossible to pass the secret file path to python-oca.

ilicmilan commented on lib/ansible/modules/cloud/opennebula/one_vm.py:
```diff
@@ -1220,8 +1223,8 @@ def get_connection_info(module):
+        module.fail_json(msg="Opennebula API url (api_url) is not specified")
     from collections import namedtuple
 
     auth_params = namedtuple('auth', ('url', 'username', 'password'))
```
Just extract `ONE_AUTH` env var and add it here in this tuple

YouhuaLi replied:
As mentioned previous comment, python-oca will do that.

ilicmilan reviewed (changes requested):
Hi @YouhuaLi, thanks for your contribution! Much appreciated.
Just add this feature with `ONE_AUTH`

felixfontein committed:
ACME: improve error handling (#49266)

* Improve error handling: when parse_json_result is False, still return the error message.

* Remove content body from info dict.

* Add changelog.

D3DeFi committed:
zabbix_hostmacro: Fixed support for LLD type user macros and added missing validate_certs attribute (#48730)

AlanCoding committed:
Fix error where pod has error and no containers (#49154)

k8s: Fix error where pod has error and no containers

felixfontein committed:
Add basic unit tests for module_utils/acme.py. (#49103)

johnboy2 committed:
ec2_vpc_nacl: Don't default name into Name unless able (#48889)

This fixes ansible/ansible#issues/48832

coreywan committed:
sets admin_password to no_log (#46604)

joaocpinto committed:
Cast int for expire field to avoid call failure to sensu API (#33239)

ryansb committed:
ec2_asg: Retry Autoscaling Group delete calls when scaling activity is in progress (#46124)

* Retry delete calls when scaling activity is in progress

* changelog

ryansb committed:
VPC peer module unhandled exception on bad peering ID (#45306)

* Improve error handling for VPC peer module when a bad peering-id is specified

* changelog

fdeschamps committed:
ignore empty lines in rabbitmqctl output (#43722)

* ignore empty lines in rabbitmqctl output

this fixes a bug with rabbitmq 3.7.5
rabbitmqctl can return empty lines, breaking the rabbitmq_parameter module
especially in a new vhost, the command rabbitmqctl list_parameters -q -p <vhost> will return an empty line

* Strip empty line in rabbitmqctl output

douglasfraser committed:
Extend DMI form factors list to match SMBIOS standard 3.2.0 (2018-04-26) (#42239)

Signed-off-by: Douglas Fraser <doug+git@idmf.net>

samdoran committed:
Improve log message (#49004)

This is the only connection plugin that doesn't state explicitly what it is in the log.

jpdasma committed:
Fix reboot plugin not working in Illumos/Solaris (#48986)

pilou- committed:
templar: ensure that exceptions are handled, fix 'AttributeError' (#48792)

* templar: ensure that exceptions are handled

* Fix AttributeError: object has no attribute 'message'

'message' attribute is deprecated since Python 2.6 and not available
with Python 3.

Simple reproducer:

    - hosts: localhost
      vars:
        not_json: "{{ 'test str' | from_json }}"
      tasks:
        - command: "echo {{ not_json }}"

spike77453 committed:
Non-existent RR returns empty string (#40680)

amybuck committed:
Update dellos10_command.py (#48167)

<!--- Your description here -->

Updated Dell OS10 to Dell EMC OS10

davidmnoriega committed:
Adding note on async+check mode incompatibility (#41427)

Fixes ansible/ansible#40991

bcoca committed:
added missing github id

Isakofff committed:
Update elb_application_lb_facts.py (#48983)

<!--- Added example how to see the facts about specific ALB -->

ryansb committed:
New module: AWS EC2 Launch Template (#46972)

* Add launch template integration tests

paneu committed:
ios_facts: Gather CDP neighbor data (#49129)

* ios_facts: Gather CDP neighbor data

* ios_facts: Create tests for ansible_net_neighbors

D3DeFi committed:
zabbix_template: fixed idempotency issues (#49188)

bcoca committed:
corrected action plugin docs (#49321)

fixes #49277

sivel committed:
Do not allow addressing an import from handlers via notify. Fixes #48936 (#49317)

YouhuaLi committed:
Merge branch 'devel' of github.com-YouhuaLi:YouhuaLi/ansible into devel

YouhuaLi commented:
this one was messed. I will create a new PR.
